Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Install Operating system and dependencies
+FROM ubuntu:24.04 AS builder
+
+RUN apt-get update 
+RUN apt-get install -y curl git wget unzip gdb libstdc++6 libglu1-mesa fonts-droid-fallback python3 yq
+RUN apt-get clean
+
+RUN mkdir /flutter_app/
+COPY . /flutter_app
+WORKDIR /flutter_app/app
+
+# download Flutter SDK from Flutter Github repo
+RUN git clone https://github.com/flutter/flutter.git -b $(echo "flutter-$(yq '.environment.flutter' /flutter_app/app/pubspec.yaml -r | cut -d'.' -f1,2)-candidate.0") /usr/local/flutter
```

Comment:
you don't need -candidate.0 or something like this, the version specified in the pubspec is exactly the tag in the github repository, currently 3.27.0

Reply:
You're right, I only searched the branches manually and not for the tags